src/openforms/validations/registry.py

```diff
@@ -118,10 +139,13 @@ def validate(self, plugin_id: str, value: str) -> ValidationResult:
                 ],
             )
 
+        SerializerClass = validator.callable.value_serializer
+        serializer = SerializerClass(data={"value": filter_empty_values(value)})
+        serializer.is_valid(raise_exception=True)
         try:
-            validator(value)
+            validator(serializer.data["value"], submission)
         except (DJ_ValidationError, DRF_ValidationError) as e:
-            errors = as_serializer_error(e)
+            errors = serializers.as_serializer_error(e)
```

Comment:
The response data shape is different for both validation stages.

HTTP 400 response now has validation error information, but the SDK is not built to handle that.
HTTP 200 response contains a top level key which designates whether the result is valid or not.

The SDK needs to be able to handle those HTTP 400s and place the validation errors with the respective (nested) fields, which is going to be... a challenge, I think. :grimacing: